WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -128,6 +131,10 @@ extension ProductFormViewModel {
     func canShareProduct() -> Bool {
         formType == .edit
     }
+
+    func canDeleteProduct() -> Bool {
+        isEditProductsRelease5Enabled && originalProduct.product.productStatus == .publish && formType == .edit
```

Comment:
any reasons we don't allow product deletion for other product statuses like draft/pending/private/custom? I've got so many draft products that I personally want to delete (from testing M4 😆 )

Reply:
Omg, You are right! Fixed